pkg/operator/controller/ingress/controller.go

```diff
@@ -134,9 +144,51 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(source.Kind(operatorCache, &configv1.Proxy{}), handler.EnqueueRequestsFromMapFunc(reconciler.ingressConfigToIngressController)); err != nil {
 		return nil, err
 	}
+	// Watch secrets in the operand namespace that are referenced as default certificates so that
+	// checkDefaultCertificate can determine whether changes to the default certificate affect upgradeability.
+	if err := c.Watch(source.Kind(operatorCache, &corev1.Secret{}), handler.EnqueueRequestsFromMapFunc(reconciler.secretToIngressController), predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.GetNamespace() == operatorcontroller.DefaultOperandNamespace
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Object.GetNamespace() == operatorcontroller.DefaultOperandNamespace
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetNamespace() == operatorcontroller.DefaultOperandNamespace
+		},
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+	}); err != nil {
+		return nil, err
+	}
 	return c, nil
 }
 
+// secretToIngressController maps a secret to a slice of reconcile requests,
+// one request per IngressController that references the secret as it's default certificate.
+func (r *reconciler) secretToIngressController(ctx context.Context, o client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+	controllers := &operatorv1.IngressControllerList{}
+	listOpts := client.MatchingFields(map[string]string{
+		"ingressDefaultCertificateName": o.GetName(),
+	})
+	if err := r.cache.List(ctx, controllers, client.InNamespace(r.config.Namespace), listOpts); err != nil {
+		log.Error(err, "failed to list ingresscontrollers for default certificate secret", "related", o.GetSelfLink())
+		return requests
+	}
+	for _, ic := range controllers.Items {
+		if operatorcontroller.RouterEffectiveDefaultCertificateSecretName(&ic, operatorcontroller.DefaultOperandNamespace).Name == o.GetName() {
```

Comment:
This `if` check should be redundant now since you used an index to get the list of items.

Reply:
Good point, done.